src/ShallowWrapper.js

```diff
@@ -119,27 +117,25 @@ class ShallowWrapper {
     if (!root) {
       this.root = this;
       this.unrendered = nodes;
-      this.renderer = createShallowRenderer();
-      withSetStateAllowed(() => {
-        performBatchedUpdates(this, () => {
-          this.renderer.render(nodes, options.context);
-          const instance = this.instance();
-          if (
-            options.lifecycleExperimental &&
-            instance &&
-            typeof instance.componentDidMount === 'function'
-          ) {
-            instance.componentDidMount();
-          }
+      this.renderer = getAdapter(options).createRenderer({ mode: 'shallow', ...options });
```

Comment:
If `mode` is being used by Enzyme internals should it be added to the `RendererOptions` type definition?